stdlib/internal/src/main/ballerina/internal/jwt_validator.bal

```diff
@@ -42,51 +42,63 @@ public type JWTValidatorConfig record {
 #            An error if token validation fails.
 public function validate(string jwtToken, JWTValidatorConfig config) returns JwtPayload|error {
     string[] encodedJWTComponents = [];
-    match getJWTComponents(jwtToken) {
-        string[] encodedJWT => encodedJWTComponents = encodedJWT;
-        error e => return e;
+    var jwtComponents = getJWTComponents(jwtToken);
+    if (jwtComponents is string[]) {
+        encodedJWTComponents = jwtComponents;
+    } else if (jwtComponents is error) {
+        return jwtComponents;
+    } else {
+        error jwtError = error(INTERNAL_ERROR_CODE, { message : "Invalid JWT token"});
+        return jwtError;
     }
 
-    JwtHeader header;
-    JwtPayload payload;
-    match parseJWT(encodedJWTComponents) {
-        error e => return e;
-        (JwtHeader, JwtPayload) result => {
-            (header, payload) = result;
-        }
+    string[] aud = [];
+    JwtHeader header = {alg: ""};
+    JwtPayload payload = {iss: "", sub: "", aud: aud, exp: 0};
+    var decodedJwt = parseJWT(encodedJWTComponents);
+    if (decodedJwt is (JwtHeader, JwtPayload)) {
+        (header, payload) = decodedJwt;
+    } else if (decodedJwt is error) {
+        return decodedJwt;
```

Comment:
Should we create a new error with INTERNAL_ERROR_CODE here?

Reply:
This error propagated from a native function in the internal package, So it has the same error code.